src/main/java/org/opensearch/knn/training/TrainingJob.java

```diff
@@ -182,6 +187,14 @@ public void run() {
                 KNNSettings.state().getSettingValue(KNNSettings.KNN_ALGO_PARAM_INDEX_THREAD_QTY)
             );
 
+            if (VectorDataType.BINARY.equals(model.getModelMetadata().getVectorDataType())) {
```

Comment:
I was just heard that this == operation is faster...
```suggestion
            if (VectorDataType.BINARY == model.getModelMetadata().getVectorDataType()) {
```